File: Michał Walencik/Task1.py
import rasterio
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Ścieżki do plików TIFF
ndvi_spring_path = "t3_ndvi2024May.tif"
ndvi_summer_path = "t3_ndvi2023_Jul_Aug.tif"
temp_spring_path = "t3_lst2024May.tif"
temp_summer_path = "t3_lst2023_Jul_Aug.tif"

# Funkcja do wczytania danych z pliku TIFF
def read_tif(filepath):
    with rasterio.open(filepath) as dataset:
        return dataset.read(1)  # Czytamy tylko pierwszą warstwę

# Wczytanie danych
ndvi_spring = read_tif(ndvi_spring_path)
ndvi_summer = read_tif(ndvi_summer_path)
temp_spring = read_tif(temp_spring_path)
temp_summer = read_tif(temp_summer_path)

# Wyświetlenie 4 obrazów
fig, axes = plt.subplots(2, 2, figsize=(12, 10))

images = [
    (ndvi_spring, "NDVI Wiosna", "Greens"),
    (ndvi_summer, "NDVI Lato", "Greens"),
    (temp_spring, "Temp. Wiosna", "coolwarm"),
    (temp_summer, "Temp. Lato", "coolwarm")
]

for ax, (data, title, cmap) in zip(axes.flat, images):
    img = ax.imshow(data, cmap=cmap)
    ax.set_title(title)
    fig.colorbar(img, ax=ax)

plt.tight_layout()
plt.show()

# Tworzenie histogramów
fig, ax = plt.subplots(1, 2, figsize=(12, 5))

sns.histplot(ndvi_spring.flatten(), bins=30, kde=True, ax=ax[0], color='green', label='NDVI Wiosna')
sns.histplot(ndvi_summer.flatten(), bins=30, kde=True, ax=ax[0], color='darkgreen', alpha=0.7, label='NDVI Lato')
ax[0].set_title('Histogram NDVI')
ax[0].legend()

sns.histplot(temp_spring.flatten(), bins=30, kde=True, ax=ax[1], color='blue', label='Temp Wiosna')
sns.histplot(temp_summer.flatten(), bins=30, kde=True, ax=ax[1], color='red', alpha=0.7, label='Temp Lato')
ax[1].set_title('Histogram Temperatury')
ax[1].legend()

plt.show()

# Wykres rozrzutu (NDVI vs Temperatura)
plt.figure(figsize=(8, 6))
sns.scatterplot(x=ndvi_spring.flatten(), y=temp_spring.flatten(), color='blue', alpha=0.5, label='Wiosna')
sns.scatterplot(x=ndvi_summer.flatten(), y=temp_summer.flatten(), color='red', alpha=0.5, label='Lato')
plt.xlabel('NDVI')
plt.ylabel('Temperatura')
plt.title('Wykres rozrzutu: NDVI vs. Temperatura')
plt.legend()
plt.show()

File: Michał Walencik/Task1.m
% File paths for the data
files = struct('ndvi_spring', 'C:\EGZ\taks1\t3_ndvi2024May.tif', ...
               'ndvi_summer', 'C:\EGZ\taks1\t3_ndvi2023_Jul_Aug.tif', ...
               'temp_spring', 'C:\EGZ\taks1\t3_lst2024May.tif', ...
               'temp_summer', 'C:\EGZ\taks1\t3_lst2023_Jul_Aug.tif');

% Function to load a TIFF file
function img = load_image(file)
    img = imread(file);  % Load the image from the file
end

% Load the data using a loop
data = struct();
fields = fieldnames(files);
for i = 1:length(fields)
    data.(fields{i}) = load_image(files.(fields{i}));
end

% Set colormap options for the plots
colormap_spring = 'spring';
colormap_temp = {'cool', 'hot'};

% Create the images and display them
figure;
subplot(2, 2, 1);
imshow(data.ndvi_spring, []); % Display NDVI Spring image
colormap(colormap_spring);
colorbar;
title('NDVI Spring');

subplot(2, 2, 2);
imshow(data.ndvi_summer, []); % Display NDVI Summer image
colormap(colormap_spring);
colorbar;
title('NDVI Summer');

subplot(2, 2, 3);
imshow(data.temp_spring, []); % Display Temperature Spring image
colormap(colormap_temp{1});
colorbar;
title('Temperature Spring');

subplot(2, 2, 4);
imshow(data.temp_summer, []); % Display Temperature Summer image
colormap(colormap_temp{2});
colorbar;
title('Temperature Summer');

% Main title for the figure
sgtitle('NDVI and Temperature Images');
set(gcf, 'Position', [100, 100, 800, 600]);  % Set the figure window size

% Plot the histograms
figure;
hold on;
% NDVI histograms
subplot(1, 2, 1);
histogram(data.ndvi_spring(:), 30, 'Normalization', 'pdf', 'FaceColor', '#00FF00', 'EdgeColor', 'black');
histogram(data.ndvi_summer(:), 30, 'Normalization', 'pdf', 'FaceColor', '#006400', 'EdgeColor', 'black');
title('NDVI Histogram');
legend('NDVI Spring', 'NDVI Summer');

% Temperature histograms
subplot(1, 2, 2);
histogram(data.temp_spring(:), 30, 'Normalization', 'pdf', 'FaceColor', '#0000FF', 'EdgeColor', 'black');
histogram(data.temp_summer(:), 30, 'Normalization', 'pdf', 'FaceColor', '#FF0000', 'EdgeColor', 'black');
title('Temperature Histogram');
legend('Temperature Spring', 'Temperature Summer');
hold off;

% Scatter plot: NDVI vs Temperature
figure;
hold on;
scatter(data.ndvi_spring(:), data.temp_spring(:), 20, 'blue', 'filled', 'MarkerFaceAlpha', 0.5);
scatter(data.ndvi_summer(:), data.temp_summer(:), 20, 'red', 'filled', 'MarkerFaceAlpha', 0.5);
xlabel('NDVI');
ylabel('Temperature');
title('Scatter Plot: NDVI vs. Temperature');
legend('Spring', 'Summer');
hold off;
